src/muse-dl.cr

```diff
@@ -65,14 +65,33 @@ module Muse::Dl
     def self.run(args : Array(String))
       parser = Parser.new(args)
 
+      delay_secs = 1
       input_list = parser.input_list
       if input_list
         File.each_line input_list do |url|
-          # TODO: Change this to nil
-          parser.reset_output_file
-          parser.url = url.strip
-          # Ask the download process to not quit the process, and return instead
-          Main.dl parser
+          begin
+            # TODO: Change this to nil
+            parser.reset_output_file
+            parser.url = url.strip
+            # Ask the download process to not quit the process, and return instead
+            Main.dl parser
+            if delay_secs >= 2
+              delay_secs /= 2
+            end
+          rescue ex : Muse::Dl::Errors::DownloadError
+            puts ex.message
+            puts ex.backtrace.join("\n    ")
+            puts "Download error. Skipping book: #{url}. Waiting for #{delay_secs} seconds before continuing."
+            # Sleep to prevent hammering the server.
+            sleep(delay_secs)
+            delay_secs *= 2
```

Comment:
If a file is broken at MUSE end, won't this get stuck forever? Maybe move on the next book if wait time gets too big?